src/web/handlers/status_list/publish_token_status.rs

```diff
@@ -0,0 +1,308 @@
+use crate::{
+    model::{StatusList, StatusListToken},
+    utils::{
+        errors::Error,
+        lst_gen::{lst_from, PublishStatus},
+        state::AppState,
+    },
+    web::handlers::status_list::error::StatusListError,
+};
+use axum::{
+    extract::{Json, State},
+    http::StatusCode,
+    response::IntoResponse,
+};
+use serde::{Deserialize, Serialize};
+use std::time::{SystemTime, UNIX_EPOCH};
+use tracing;
+
+// Request payload for publishing a status list token
+#[derive(Deserialize, Serialize, Clone)]
+pub struct PublishTokenStatusRequest {
+    pub list_id: String,
+    pub updates: Vec<PublishStatus>,
+    #[serde(default)]
+    pub sub: Option<String>,
+    #[serde(default)]
+    pub ttl: Option<String>,
+    pub bits: u8,
+}
+
+// Handler to create a new status list token
+pub async fn publish_token_status(
+    State(appstate): State<AppState>,
+    Json(payload): Json<PublishTokenStatusRequest>,
+) -> Result<impl IntoResponse, StatusListError> {
+    // Ensure the repository is available
+    let store = appstate
+        .repository
+        .as_ref()
+        .ok_or(StatusListError::InternalServerError)?;
+
+    // Validate that bits is one of the allowed values (1, 2, 4, 8)
+    if ![1, 2, 4, 8].contains(&payload.bits) {
+        return Err(StatusListError::UnsupportedBits);
+    }
+
+    // Generate the compressed status list; use empty encoding if no updates
+    let lst = if payload.updates.is_empty() {
+        base64url::encode([])
+    } else {
+        lst_from(payload.updates, payload.bits as usize).map_err(|e| {
+            tracing::error!("lst_from failed: {:?}", e);
+            match e {
+                Error::Generic(msg) => StatusListError::Generic(msg),
+                Error::InvalidIndex => StatusListError::InvalidIndex,
+                Error::UnsupportedBits => StatusListError::UnsupportedBits,
+                _ => StatusListError::Generic(e.to_string()),
+            }
+        })?
+    };
+
+    // Check for existing token to prevent duplicates
+    match store
+        .status_list_token_repository
+        .find_one_by(payload.list_id.clone())
+        .await
+    {
+        Ok(Some(_)) => {
+            tracing::info!("Status list {} already exists", payload.list_id);
+            Err(StatusListError::StatusListAlreadyExists)
+        }
+        Ok(None) => {
+            // Calculate issuance and expiration timestamps
+            let iat = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_secs() as i64;
+            let exp = payload.ttl.as_ref().and_then(|ttl| {
```

Comment:
```suggestion
            let exp = match payload.ttl.as_ref() {
                 Some(ttl) => match ttl.parse::<i64>() {
                           Ok(ttl_seconds) => Some(iat.saturating_add(ttl_seconds)),
                           Err(_) => return Err(StatusListError::Generic("Invalid TTL format".to_string())),
                 },
                 None => None,
            };
```

You can also add error handling for invalid `TTL` values instead of silently failing